hack/make/.build-empty-images

```diff
@@ -24,3 +29,27 @@ if ! docker image inspect emptyfs > /dev/null; then
 	)
 	rm -rf "$dir"
 fi
+
+# without c8d image store, image id is the config's id
+dangling_cfg=0df1207206e5288f4a989a2f13d1f5b3c4e70467702c1d5d21dfc9f002b7bd43
+# with c8d image store, image id is the id of manifest/manifest list.
+dangling_mfst=16d365089e5c10e1673ee82ab5bba38ade9b763296ad918bd24b42a1156c5456
+if imageNotPresent "$dangling_cfg" && imageNotPresent "$dangling_mfst"; then
+	dir="$DEST/dangling"
+	mkdir "$dir"
+	(
+		cd "$dir"
+		echo -n '{"schemaVersion":2,"manifests":[{"mediaType":"application/vnd.docker.distribution.manifest.v2+json","digest":"sha256:16d365089e5c10e1673ee82ab5bba38ade9b763296ad918bd24b42a1156c5456","size":264,"annotations":{"org.opencontainers.image.created":"2023-05-19T08:00:44Z"},"platform":{"architecture":"amd64","os":"linux"}}]}' > index.json
+		echo -n '[{"Config":"blobs/sha256/0df1207206e5288f4a989a2f13d1f5b3c4e70467702c1d5d21dfc9f002b7bd43","RepoTags":null,"Layers":null}]' > manifest.json
+		mkdir -p blobs/sha256
+		echo -n '{"schemaVersion":2,"mediaType":"application/vnd.docker.distribution.manifest.v2+json","config":{"mediaType":"application/vnd.docker.container.image.v1+json","digest":"sha256:0df1207206e5288f4a989a2f13d1f5b3c4e70467702c1d5d21dfc9f002b7bd43","size":390},"layers":[]}' > blobs/sha256/$dangling_mfst
+		echo -n '{"architecture":"amd64","config":{"Env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"],"WorkingDir":"/","Labels":{"org.mobyproject.test.specialimage":"1"},"OnBuild":null},"created":null,"history":[{"created_by":"LABEL org.mobyproject.test.specialimage=1","comment":"buildkit.dockerfile.v0","empty_layer":true}],"os":"linux","rootfs":{"type":"layers","diff_ids":null}}' > blobs/sha256/$dangling_cfg
```

Comment:
Is it possible to compute these SHAs in the script from the inline JSON, instead of using them as magic numbers?

e.g. we have the resolution of bash here, so it should be pretty easy to store these values in variables and hash them using herestrings, unless I'm missing something trivial.

Reply:
I'm personally good with hard-coding it; it also makes it easier to grep/search for these.

Fun fact; the OCI image-spec even has a hard-coded digest for "empty image"; https://github.com/opencontainers/image-spec/blob/0b40f0f367c396cc5a7d6a2e8c8842271d3d3844/specs-go/v1/manifest.go#L43-L49

Reply:
Fair enough

Reply:
NOOOOOOOOOO the scratch descriptor is not for "empty image" (it's not a valid tar) - it's for "my artifact doesn't require config or layers" (and is meant as a placeholder type to satisfy the "config is required, layers must have at least one item")

Reply:
From the images I maintain in https://explore.ggcr.dev/?repo=tianon/scratch, what you really want (instead of the OCI's scratch descriptor) is what's in `tianon/scratch:latest`, `tianon/scratch:nolayers`, `tianon/scratch:minimalconfig`, `tianon/scratch:oci`, or `tianon/scratch:*zerobytes` :sweat_smile:

(I haven't looked at the details of this PR yet, but want to make sure it's very very clear that the OCI's "scratch descriptor" is not relevant here :speak_no_evil:)

Reply:
The scratch descriptor is not used there at all. What this PR needs is just _some_ image without any tag/reference, and I made it a layerless image just because it's easy to construct by hand.